### PR TITLE
PAF-27: create UI integration tests

### PR DESCRIPTION
**Pull request by Rhodine-orleans-lindsay**
## What
- Create initial UI integration tests

## Why
- To allow for testing in pipeline

### CODE REVIEW

File: test/_ui-integration/paf/application.spec.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    testApp = getSupertestApp(SUBAPP);
+    passStep = testApp.passStep;
+    initSession = testApp.initSession;
+  });
+
+  it('goes to the crime-type page', async () => {
+    const URI = '/start';
+    await initSession(URI);
+    const response = await passStep(URI, {});
+    expect(response.text).to.contain('Found. Redirecting to /paf/crime-type');
```
Comment (sulthan-ahmed) [inline test/_ui-integration/paf/application.spec.js:18]: Can you explain why it says 'Found. Redirecting...' where is that response from? is it from the response body?

Comment (Rhodine-orleans-lindsay) [reply test/_ui-integration/paf/application.spec.js:18]: Yes this comes from the response.  The expected http status 302 Found happens on a calling
passStep() successfully - it redirects to the next page

Comment (sulthan-ahmed) [reply test/_ui-integration/paf/application.spec.js:18]: ah ok, why does it redirect on every page?

File: test/helpers/supertest_session/session-data/base.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          /* eslint-disable no-nested-ternary */
+          if (overridenPages && require(`./${appName}/${overridenPages}${prop}`)) {
+            Object.assign(props, require(`./${appName}/${overridenPages}${prop}`),
+              data ? data : typeof stepOrData === 'object' ? stepOrData : {});
+          } else {
+            Object.assign(props, require(`./${appName}/pages${prop}`),
+              data ? data : typeof stepOrData === 'object' ? stepOrData : {});
+          }
+
+          /* eslint-disable no-empty */
```
Comment (sulthan-ahmed) [inline test/helpers/supertest_session/session-data/base.js:26]: I would do then you don't need it twice
```suggestion
          /* eslint-disable no-empty */
```

File: test/_ui-integration/paf/application.spec.js (hunk 1 above)
Comment (sulthan-ahmed) [reply test/_ui-integration/paf/application.spec.js:18]: acceptance tests you usually have to tell it to continue, so I'm wondering how this work

Comment (Rhodine-orleans-lindsay) [reply test/_ui-integration/paf/application.spec.js:18]: the passStep function calls a postUrl function (see the supertest-utilities.js file) which basically acts as the 'continue'/ 'submit' in acceptance tests

File: test/helpers/supertest_session/session-data/base.js (hunk 2 above)
Comment (Rhodine-orleans-lindsay) [reply test/helpers/supertest_session/session-data/base.js]: I have amended this now

File: test/_ui-integration/paf/application.spec.js (hunk 1 above)
Comment (sulthan-ahmed) [reply test/_ui-integration/paf/application.spec.js:18]: Ok thanks Rhodine for the explanation, I understand some of it but sounds like I should run through it in debug mode to fully get it.  Nice work

**Review by sulthan-ahmed — APPROVED** [review]
lgtm

**Pull request merged**